scipy/stats/_distn_infrastructure.py

```diff
@@ -2386,18 +2445,48 @@ def fit(self, data, *args, **kwds):
         loc = kwds.pop('loc', start[-2])
         scale = kwds.pop('scale', start[-1])
         args += (loc, scale)
-        x0, func, restore, args = self._reduce_func(args, kwds)
+        x0, func, restore, args = self._reduce_func(args, kwds, data=data)
         optimizer = kwds.pop('optimizer', optimize.fmin)
         # convert string to function in scipy.optimize
         optimizer = _fit_determine_optimizer(optimizer)
         # by now kwds must be empty, since everybody took what they needed
         if kwds:
             raise TypeError("Unknown arguments: %s." % kwds)
 
-        vals = optimizer(func, x0, args=(ravel(data),), disp=0)
+        # method of moments could be done with fsolve/root instead of an
+        # optimizer, but using an optimizer reduces maintains the structure of
+        # MM/MLE and paves the way for Generalized Method of Moments
+        vals, obj = optimizer(func, x0, args=(ravel(data),), disp=0,
+                              full_output=True)[:2]
```

Comment:
This breaks backwards compatibility.  The `fit` docstring says that, in effect, the signature of the optimizer must be `optimizer(func, x0, args=(), disp=0)`.   If a user has passed in a callable with that signature, this call will raise a `TypeError`.

You're only using `full_output` to get the value at the minimum.  Instead of using `full_output`, could you  evaluate `func` after the call to `optimizer`?

Reply:
Yes.